go/cmd/vtgate/vtgate.go

```diff
@@ -68,8 +136,22 @@ func main() {
 				log.Errorf("unknown tablet type: %v", ttStr)
 				continue
 			}
-			tabletTypes = append(tabletTypes, tt)
+			if tt == topodatapb.TabletType_MASTER || tt == topodatapb.TabletType_REPLICA || tt == topodatapb.TabletType_RDONLY {
```

Comment:
I'm not sure about adding this check here. Right now we don't restrict the tablet type. What is the reason to do it?

Reply:
> I took the liberty of suggesting slightly different messages.
> Also, could you use `vterrors` instead of `fmt.Errorf`?

see, will use vterrors

Reply:
> I'm not sure about adding this check here. Right now we don't restrict the tablet type. What is the reason to do it?

i am just wondering at current situation if this three kind of types just are making sense ? as i know all other tablet type are supposed not providing services ?

Reply:
I think if we want to restrict it, it should match
https://github.com/vitessio/vitess/blob/master/go/vt/vttablet/tabletserver/tabletserver.go#L384

`BATCH` is same as `RDONLY` and I don't know if anyone uses `EXPERIMENTAL`.

Reply:
understood. will modify